### PR TITLE
RPKM app that will work in pipeline

### PR DESCRIPTION
**Pull request by aledj2**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,44 @@
...
+
+
+## What data are required for this app to run?
+* A BED file for the CNV regions 
+  * This bedfile must NOT have a header
+  *  Each entry must have  'chr' removed eg:
+      `11    108093508    108093963    ATM_Ex01_PM`
+  *  An app (RPKM_bedfile) has been created which can be used to convert the data.bed bed file produced by mokabed into the desired format (RPKM.bed)
+* A project in which to find BAM files
+  * BAM files must be found in /output/
```
Comment (woook) [inline README.md]: Is this correct? Is the output folder in the root of the project?

File: README.md [hunk 2]
```diff
@@ -1 +1,44 @@
...
+* The variant calling bed file Pan number
+  * The RPKM bed file will have a different panel number (for +/-50bp) therefore the panel number which corresponds to the +/-10bp panel is required to download the relevant BAM files.
```
Comment (woook) [inline README.md]: Why is the RPKM bed file different to the coverage bed file?

File: README.md [hunk 3]
```diff
@@ -1 +1,44 @@
...
+
+
+## What does this app output?
+Within a folder named conifer_output a folder is created for that Pan number. Within this folder the following files are created:
+* A text file for each BAM input
+* summary.txt - A file with a column containing the RPKM value for each sample
+* combined_bed_summary.txt - The summary.txt file merged with the bedfile to give the coordinates for each RPKM value.
+
+## How does this app work?
+This app is based on the code in the github repo https://github.com/moka-guys/RPKM.
```
Comment (woook) [inline README.md]: ?
https://github.com/moka-guys/dnanexus_RPKM

**Review by woook — CHANGES_REQUESTED** [review]
I've some queries - attached to lines in the readme

File: README.md (hunk 1 above)
Comment (aledj2) [inline README.md]: yes that's the path to the bam files. The dx download command ensures it's the non-deduplicated bams that are used.

File: README.md (hunk 2 above)
Comment (aledj2) [inline README.md]: In moka there is entry in the gene panel table (which creates the Pannumber) for +/-10 and another for +/-50 which also details the control sites for CNV analysis.

File: README.md (hunk 3 above)
Comment (aledj2) [inline README.md]: dnanexus_RPKM is this app. the RPKM code is in the repo stated above (so it can be run standalone)

File: README.md (hunk 1 above)
Comment (woook) [inline README.md]: As mentioned elsewhere - it makes no difference whether you use the BAMs straight from BWA or whether you use the _*duplicate-marked*_ BAMs

File: README.md (hunk 2 above)
Comment (woook) [inline README.md]: But why are we counting reads that fall outside the region of interest?

**Review by woook — DISMISSED** [review]

**Review by woook — CHANGES_REQUESTED** [review]
See in-line comments

File: README.md (hunk 1 above)
Comment (aledj2) [inline README.md]: Will add that to Readme

File: README.md (hunk 2 above)
Comment (aledj2) [inline README.md]: That's what the scientists asked for...

File: README.md [hunk 4]
```diff
@@ -1 +1,45 @@
...
+  * This bedfile must NOT have a header
+  *  Each entry must have  'chr' removed eg:
+      `11    108093508    108093963    ATM_Ex01_PM`
+  *  An app (RPKM_bedfile) has been created which can be used to convert the data.bed bed file produced by mokabed into the desired format (RPKM.bed)
+* A project in which to find BAM files
+  * BAM files must be found in /output/
+  * Either deduplicated or 'pre-processed' BAM files can be used (the app downloads the preprocessed bam file)
+* The variant calling bed file Pan number
+  * The RPKM bed file will have a different panel number (for +/-50bp) therefore the panel number which corresponds to the +/-10bp panel is required to download the relevant BAM files.  
+
```
Comment (woook) [inline README.md]: Please update readme in light of move to using one panel number for both coverage and CNV analysis.

File: src/code.sh [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+
+
+#make test dir
+mkdir to_test
+cd to_test
+echo $project_name
+dx download $project_name:output/*$pannum*001.ba* --auth $API_KEY
+filecount="$(ls *"$pannum"* | grep . -c)"
+if (( $filecount < 6 )); then
```
Comment (woook) [inline src/code.sh]: Will RPKM generate solid number with just the reference samples? Do we / do we need to take into account the sex of the samples for X genes?

**Review by woook — APPROVED** [review]
Technically the code is ok but there are doubts regarding the validity of the output for CNV dosage quotient analysis.

**Pull request merged**